src/main/scala/scalaz/analytics/LocalAnalyticsModule.scala

```diff
@@ -0,0 +1,179 @@
+        override def diff: (A, A) =>: A = RowFunction.Diff(Type[A].reified)
+        override def mod: (A, A) =>: A  = RowFunction.Mod(Type[A].reified)
+      }
+  }
+
+  implicit override val unknown: Type[Unknown] = new Type[Unknown] {
+    override def reified: Reified = Reified.Unknown
+  }
+  implicit override val intType: Type[scala.Int]   = LocalType(Reified.Int)
+  implicit override val intNumeric: Num[scala.Int] = LocalNum[Int]
+
+  implicit override val longType: Type[scala.Long]   = LocalType(Reified.Long)
+  implicit override val longNumeric: Num[scala.Long] = LocalNum[scala.Long]
+
+  implicit override val floatType: Type[scala.Float]   = LocalType(Reified.Float)
+  implicit override val floatNumeric: Num[scala.Float] = LocalNum[scala.Float]
+
+  implicit override val doubleType: Type[scala.Double]   = LocalType(Reified.Double)
+  implicit override val doubleNumeric: Num[scala.Double] = LocalNum[scala.Double]
+
+  implicit override def tuple2Type[A: Type, B: Type]: Type[(A, B)] = new Type[(A, B)] {
+    override def reified: Reified = Reified.Tuple2(LocalType.typeof[A], LocalType.typeof[B])
+  }
+
+  /**
+   * A typeclass that produces a Reified
+   */
+  sealed trait LocalType[A] {
+    def reified: Reified
+  }
+
+  object LocalType {
+    def typeof[A](implicit ev: LocalType[A]): Reified = ev.reified
+
+    private[LocalAnalyticsModule] def apply[A](r: Reified): Type[A] =
+      new Type[A] {
+        override def reified: Reified = r
+      }
+  }
+
+  /**
+   * The set of reified types.
+   * These represent all the Types that scalaz-analytics works with
+   */
+  sealed trait Reified
+
+  object Reified {
+
+    case object Int                           extends Reified
+    case object Long                          extends Reified
+    case object Float                         extends Reified
+    case object Double                        extends Reified
+    case object String                        extends Reified
+    case object Decimal                       extends Reified
+    case object Unknown                       extends Reified
+    case class Tuple2(a: Reified, b: Reified) extends Reified
+  }
+
+  /**
+   * A reified DataStream program
+   */
+  sealed trait LocalDataStream
+
+  object LocalDataStream {
+    case class Empty(rType: Reified)                             extends LocalDataStream
+    case class Union(a: LocalDataStream, b: LocalDataStream)     extends LocalDataStream
+    case class Intersect(a: LocalDataStream, b: LocalDataStream) extends LocalDataStream
+    case class Except(a: LocalDataStream, b: LocalDataStream)    extends LocalDataStream
+    case class Distinct(a: LocalDataStream)                      extends LocalDataStream
+    case class Map[A, B](d: LocalDataStream, f: A =>: B)         extends LocalDataStream
+    case class Sort(a: LocalDataStream)                          extends LocalDataStream
+    case class DistinctBy[A, B](d: LocalDataStream, f: A =>: B)  extends LocalDataStream
```

Comment:
Will remove the type params.

Do we need all of `Distinct`/`DistinctBy`/`GroupBy` anyway?
I think `DistinctBy` = `GroupBy` + `Distinct` right?

Reply:
We should minimize the number of cases and provide helper functions for compositions. That way it's easier to implement module.

Reply:
I'm going to leave this as part of #4 because we need to work out the semantics of our `DataStream`/`Dataset` representation anyway (e.g. `DistinctBy` doesn't make sense on a stream without windows)